### PR TITLE
[Api][SIMS#925]pir application error message bug fix

### PR DESCRIPTION
**Pull request by henryoforeh-aot**
I fixed the issue where the studyStartDate and studyEndDate attributes of the payload object (from formio) in the completeProgramInfoRequest method of the program-info-request-controller were undefined. This was sending undefined values for payload.studyStartDate and payload.studyEndDate in the call to validateOverlappingDatesAndPIR.

image

### CODE REVIEW

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 1]
```diff
@@ -281,14 +281,17 @@ export class ProgramInfoRequestController extends BaseController {
         } as EducationProgramOffering;
       }
 
+      const studyStartDate = new Date(payload.studentStudyStartDate);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:284]: I believe that if the offering is present (line 266) the studentStudyStartDate and studentStudyEndDate must come from the offering and not the payload.

Comment (henryoforeh-aot) [reply sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:284]: I can see from the code further down that the offering might still need the studyStartDate and studyEndDate to come from the payload which will still be undefined.

image

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 2]
```diff
@@ -248,47 +248,46 @@ export class ProgramInfoRequestController extends BaseController {
...
-        const offering = await this.offeringService.getOfferingById(
-          payload.selectedOffering,
-        );
-        selectedOfferingIntensity = offering.offeringIntensity;
-      }
-
       this.applicationService.checkOfferingIntensityMismatch(
         application.data.howWillYouBeAttendingTheProgram,
-        selectedOfferingIntensity,
+        payload.offeringIntensity,
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:253]: I think you need this offeringIntensity logic
image

Comment (ann-aot) [reply sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:253]: get offeringIntensity from offering object `offeringLocation` line 262. and move this function after line 280

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 3]
```diff
@@ -248,47 +248,46 @@ export class ProgramInfoRequestController extends BaseController {
...
+        studyStartDate = offeringLocation.studyStartDate;
+        studyEndDate = offeringLocation.studyEndDate;
         // Offering exists, is valid and just need to be associated
         // with the application to complete the PIR.
         offeringToCompletePIR = {
           id: payload.selectedOffering,
         } as EducationProgramOffering;
       }
 
+      console.log("payload", payload);
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:281]: please remove console

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 4]
```diff
@@ -306,6 +305,7 @@ export class ProgramInfoRequestController extends BaseController {
         );
       }
     } catch (error) {
+      console.log("error PIR", error);
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:308]: pls remove console

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 5]
```diff
@@ -248,47 +248,46 @@ export class ProgramInfoRequestController extends BaseController {
...
       if (payload.selectedOffering) {
         // Check if the offering belongs to the location.
-        const offeringLocationId =
+        const offeringLocation =
           await this.offeringService.getOfferingLocationId(
             payload.selectedOffering,
           );
-        if (offeringLocationId !== locationId) {
+
+        if (offeringLocation.institutionLocation.id !== locationId) {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:263]: I believe that there is a missing check to verify if the _offeringLocation_ exists.
For now maybe we can just mark the _offeringLocation_ as possible nullable as below:
```
if (offeringLocation?.institutionLocation.id !== locationId)
```

**Review by ann-aot — COMMENTED** [review]
Looks Good @henryoforeh-aot. Please address the comments added by @andrewsignori-aot

File: sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts [hunk 6]
```diff
@@ -305,15 +305,20 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
...
    */
-  async getOfferingLocationId(offeringId: number): Promise<number> {
-    const locationIdQuery = await this.repo
+  async getOfferingLocationId(
+    offeringId: number,
+  ): Promise<EducationProgramOffering> {
+    return this.repo
       .createQueryBuilder("offerings")
+      .select([
+        "location.id",
```
Comment (guru-aot) [inline sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts:314]: 👍

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @henryoforeh-aot

**Review by ann-aot — APPROVED** [review]
LGTM 👍 Thanks for doing the changes @henryoforeh-aot

**Review by dheepak-aot — APPROVED** [review]
👍

**Pull request merged**